bmc-acf/acf_manager.cpp

```
@@ -0,0 +1,461 @@
+                                     timeSinceUnixEpocInSecondsParm);
+        if (CeLogin::CeLoginRc::Success != sRc)
+        {
+            log<level::ERR>("ACF time expired");
+        }
+    }
+    if (CeLogin::CeLoginRc::Success == sRc)
+    {
+        // YYYY-MM-DD
+        char sTimeStr[20];
+        sprintf(sTimeStr, "%04u-%02u-%02u", sJsonData->mExpirationDate.mYear,
+                sJsonData->mExpirationDate.mMonth,
+                sJsonData->mExpirationDate.mDay);
+        sDate = sTimeStr;
+    }
+
+    if (sDecodedAsn)
+        CELoginSequenceV1_free(sDecodedAsn);
+    if (sJsonData)
+        OPENSSL_free(sJsonData);
+    return sRc;
+}
+
+std::tuple<std::vector<uint8_t>, bool, std::string>
+    ACFCertMgr::installACF(std::vector<uint8_t> accessControlFile)
+{
+    CeLogin::CeLoginRc sRc = CeLogin::CeLoginRc::Failure;
+    bool isAcfInstalled = false;
+    std::string sDate;
+    bool prodKeyExists = false;
+    bool devKeyExists = false;
+    // delete acf file if accessControlFile is empty
+    if (accessControlFile.empty())
+    {
+        try
+        {
+            isAcfInstalled = std::filesystem::exists(ACF_FILE_PATH);
+        }
+        catch (std::filesystem::filesystem_error& e)
+        {
+            log<level::ERR>("Filesystem error");
+        }
+
+        if (isAcfInstalled)
+        {
+            try
+            {
+                std::filesystem::remove(ACF_FILE_PATH);
+                isAcfInstalled = false;
+                log<level::INFO>("Removed ACF");
+            }
+            catch (std::filesystem::filesystem_error& e)
+            {
+                log<level::ERR>("Filesystem error");
+            }
+        }
+        return std::make_tuple(accessControlFile, isAcfInstalled, sDate);
+    }
+
+    try
+    {
+        prodKeyExists = std::filesystem::exists(PROD_PUB_KEY_FILE_PATH);
+        devKeyExists = std::filesystem::exists(DEV_PUB_KEY_FILE_PATH);
+    }
+    catch (std::filesystem::filesystem_error& e)
```

catch all exceptions by const